### PR TITLE
Client Server communication through tRPC

### PR DESCRIPTION
**Pull request by piotrwitek**
Implemented serverless lamba handling client requests using tRPC client.
Added SDK infra provisioning using SSt stack.
Added tests to confirm serialization/deserialization during client-server communication is working correcly.

### CODE REVIEW

File: apps/summerfi-api/lib/sdk-router-function/e2e/refinance-e2e.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable turbo/no-undeclared-env-vars */
+import { createTRPCClient, httpBatchLink } from '@trpc/client'
+import { Protocols, type Position, type PositionId, type Protocol } from '../src/sdk-mocks'
```
Comment (robercano) [inline apps/summerfi-api/lib/sdk-router-function/e2e/refinance-e2e.ts:3]: With the separated tsconfig and tsconfig.build version, you can now import with `~sdk/sdk-mocks`

File: apps/summerfi-api/lib/sdk-router-function/src/server.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Server
+ */
+
+const t = initTRPC.create()
+export const router = t.router
+export const publicProcedure = t.procedure
+export const createCallerFactory = t.createCallerFactory
+
+export const appRouter = router({
```
Comment (robercano) [inline apps/summerfi-api/lib/sdk-router-function/src/server.ts:21]: Not necessary now, but moving forward we should think of a way of organizing this functions. Perhaps in different files and linked them here, for example

**Review by robercano — DISMISSED** [review]
Looks good to me!

File: apps/summerfi-api/lib/sdk-router-function/e2e/refinance-e2e.ts (hunk 1 above)
Comment (piotrwitek) [reply apps/summerfi-api/lib/sdk-router-function/e2e/refinance-e2e.ts:3]: thanks done

File: apps/summerfi-api/lib/sdk-router-function/src/server.ts (hunk 2 above)
Comment (piotrwitek) [reply apps/summerfi-api/lib/sdk-router-function/src/server.ts]: I updated structure a bit to be better organized let me WDYT :)

**Review by robercano — APPROVED** [review]
Looks good!

**Pull request merged**